src/adapter/http/src/simple_protocol/handlers.rs

```diff
@@ -138,6 +138,7 @@ pub async fn dataset_data_put_handler(
     axum::TypedHeader(content_length): axum::TypedHeader<axum::headers::ContentLength>,
     body_stream: axum::extract::BodyStream,
 ) -> Result<(), ApiError> {
+    #[cfg_attr(target_pointer_width = "64", allow(clippy::cast_possible_truncation))]
```

Comment:
I think `size_hint` should be `u64` everywhere instead. FILE apis return `u64` for size.
The only place where we break it is a test: 
   `test_insert_stream_long`
where it is mapped to container size - there it is safe to cast `usize` to `u64`, and it should not create any linting warning.

Reply:
Done